Source/Particles/Gather/GetExternalFields.H

```diff
@@ -151,7 +151,10 @@ struct GetExternalEBField
             // This assumes that vzp > 0.
             amrex::ParticleReal const zl_bounded = std::min(std::max(zl, lens_start), lens_end);
             amrex::ParticleReal const zr_bounded = std::min(std::max(zr, lens_start), lens_end);
-            amrex::ParticleReal const frac = (zr_bounded - zl_bounded)/(zr - zl);
+            amrex::ParticleReal frac=1.;
```

Comment:
If the particle is at rest, in the plasma lens, then the residence correction should be 1. (The particle is feeling the force during the full timestep.)

Reply:
How about the ternary

amrex::ParticleReal const frac = ((zr - zl) == 0._rt ? 1._rt : (zr_bounded - zl_bounded)/(zr - zl));